### PR TITLE
[METADATA]Create json file containing last modified dates for data sources

### PR DESCRIPTION
**Pull request by HAEKADI**
related to https://github.com/annuaire-entreprises-data-gouv-fr/search-api/issues/407

PR strategy:

- For each data source, a metadata file is created in the same Minio folder, recording the last modified date. This date is determined during the preprocessing phase of each source in its respective DAG. Data sources downloaded from data.gouv use the data.gouv API to retrieve the last modified date. Other sources rely on available HTTP header information for the last modified date. Special cases, like RNE, extract the date from the latest database file name, while Insee uses the date of the most recent API call.

- Once all sources have their respective metadata files, a task in the ETL DAG consolidates these into a single general metadata file, which is then saved in Minio (see example below). To ensure consistency and avoid exposing incomplete data, this consolidated file is only moved to the latest folder for API exposure after successful indexing has been confirmed.
Screenshot 2024-10-09 at 13 53 42

### CODE REVIEW

File: workflows/data_pipelines/convcollective/task_functions.py [hunk 1]
```diff
@@ -85,6 +93,12 @@ def send_file_to_minio():
                 "dest_path": "convention_collective/new/",
                 "dest_name": "cc.csv",
             },
+            {
+                "source_path": CC_TMP_FOLDER,
+                "source_name": "metadata.json",
+                "dest_path": "convention_collective/new/",
+                "dest_name": "metadata.json",
+            },
```
Comment (XavierJp) [inline workflows/data_pipelines/convcollective/task_functions.py:101]: I feel like a large part of this PR could be factorize with a helper : 
`get_minio_metadata_file(CC_TMP_FOLDER, "convention_collective/new/")`
that returns 
```
{
                "source_path": CC_TMP_FOLDER,
                "source_name": "metadata.json",
                "dest_path": "convention_collective/new/",
                "dest_name": "metadata.json",
            },
```

File: workflows/data_pipelines/ess_france/task_functions.py [hunk 2]
```diff
@@ -29,21 +35,28 @@ def send_file_to_minio():
...
 
 
 def compare_files_minio():
-    is_same = minio_client.compare_files(
+    are_files_identical = minio_client.compare_files(
         file_path_1="ess/new/",
         file_name_2="ess_france.csv",
         file_path_2="ess/latest/",
         file_name_1="ess_france.csv",
     )
```
Comment (XavierJp) [inline workflows/data_pipelines/ess_france/task_functions.py:54]: the compare file to minio looks always the same. Cant it be factorize ?

**Review by XavierJp — APPROVED** [review]
huge work, congrats. A bit hard to read but I know this feature was not easy. I would say a good approach for this kind of code (but it might not work well with airflow specific constraints) is to have every sources inheriting the same class. Have this class implement some get_last_modified_date logic and then override this logic for particular cases

**Comment by HAEKADI** [comment]
The complexity of the codebase is a valid concern. As we discussed, I will go ahead and merge this PR as it stands, and I'll focus on a major refactor in the next phase, which is outlined here: https://github.com/annuaire-entreprises-data-gouv-fr/search-infra/issues/405.

**Pull request merged**